sdk/storage/azure-storage-files-shares/src/share_responses.cpp

```diff
@@ -60,4 +61,115 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     };
   }
 
+  void ListSharesPagedResponse::OnNextPage(const Azure::Core::Context& context)
+  {
+    auto protocolLayerOptions = _detail::ShareRestClient::Service::ListSharesSinglePageOptions();
+    protocolLayerOptions.IncludeFlags = m_operationOptions.ListSharesIncludeFlags;
+    protocolLayerOptions.ContinuationToken = NextPageToken;
+    protocolLayerOptions.MaxResults = m_operationOptions.PageSizeHint;
```

Comment:
@JeffreyRichter wouldn't all `PagedResponse<T>` types require the `PageSizeHint` and `ContinuationToken` options? Did we consider introducing a `PagedResponseOptions` derived class in Azure.Core to enforce that and avoid having each options class having to define the same two options (similar to `ClientOptions`)?

Reply:
MaxResults may have different meanings. In this example, it means the maximum number entries in response body. While in GetPageRanges it means the maximum size of HTTP response body, so it should have a different name like `MaxBodySize`.

Reply:
@ahsonkhan: No, not all service operations accept page size hints or use continuation tokens.